src/widgets/virtual_list.rs

```diff
@@ -0,0 +1,309 @@
+            VirtualListItem { index, mounted }
+        };
+        if self.items.is_empty() || first.unwrap() > end_item || last.unwrap() < start_item {
+            self.items.clear();
+            self.items.extend((start_item..=end_item).map(closure));
+        } else {
+            let first = first.expect("List is not empty");
+            let last = last.expect("List is not empty");
+            if first < start_item {
+                while self.items.front().is_some() && self.items.front().expect("Checked is some").index < start_item {
+                    self.items.pop_front();
+                }
+            }
+            if last > end_item {
+                while self.items.back().is_some() && self.items.back().expect("Checked is some").index > end_item {
+                    self.items.pop_back();
+                }
+            }
+            // no extend front :(
+            for item in (start_item..first).rev() {
+                self.items.push_front(closure(item));
+            }
+            self.items.extend(((last+1)..=end_item).map(closure));
+        }
+
+        let item_size = Size::new(width, item_height_upx);
+        let constraint = item_size.map(ConstraintLimit::Fill);
+
+        for item in &self.items {
+            context.for_other(&item.mounted).layout(constraint);
+        }
+
+        let item_size = item_size.into_signed();
+        let scroll = self.scroll.get_tracking_invalidate(context).into_signed();
+
+        for item in &self.items {
+            context.set_child_layout(
+                &item.mounted,
+                Rect::new(
+                    Point::new(Px::ZERO, (item_height_upx * item.index as f32).into_signed() - scroll.y),
+                    item_size,
+                )
+            );
+        }
+        
+        self.control_size.set(new_control_size);
+        self.content_size.set(Size::new(width, content_height));
+
+        new_control_size
+    }
+
+    fn mouse_wheel(
+        &mut self,
+        _device_id: DeviceId,
+        delta: MouseScrollDelta,
+        _phase: TouchPhase,
+        context: &mut EventContext<'_>,
+    ) -> EventHandling {
+        let mut handled = false;
+        {
+            let mut vertical = self.vertical_scroll.expect_made().widget().lock();
+            handled |= vertical
+                .downcast_mut::<ScrollBar>()
+                .expect("a ScrollBar")
+                .mouse_wheel(delta, context)
+                .is_break();
+        }
+        if handled {
+            self.show_scrollbars(context);
+            context.set_needs_redraw();
+
+            HANDLED
+        } else {
+            IGNORED
+        }
+    }
+}
+
+fn constrain_child(constraint: ConstraintLimit, measured: UPx) -> UPx {
+    match constraint {
+        ConstraintLimit::Fill(size) => size.min(measured),
+        // change from Scroll widget: returning just measured here would break the functionality (render too many items)
```

Comment:
If this function always calls min, we should just remove this function and call `constraint.min(measured)`.

Reply:
I'm not sure what's the correct solution here - this is mostly based on the scroll widget which uses this function in the original way. If that min constraint/size is the correct behavior, we can simplify it then.

Reply:
I think the proper way for this widget to be have is to ask the child to layout is to tell it to SizeToFit within the size expected for each row. If the size reported is less, the widget should be given less space oriented in the top-left of the row. Additionally, the row should never be allowed to be larger than the expected cell size.